docs/reduction-in-locks.md

```diff
@@ -0,0 +1,107 @@
+2. Copy `non-InnoDB` files.
+3. Check the file operations that were tracked and recopy the tablespaces.
+4. Create additional `metadata` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This is particularly important for streaming backups.
+5. Gather a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+6. Stop the redo thread once it copies at least up to sync point at step 4.
+7. Release the backup lock on the server.
+
+??? note "The list of metadata files"
+
+    | File         | Description                                                                                                      |
+    |--------------|------------------------------------------------------------------------------------------------------------------|
+    | `.new`       | For new files that are created while the backup in progress and for files that need to be recopied due to encryption or `ADD INDEX` operations.                                  |
+    | `.del`       | For deleted files. For a file that has been copied by backup but deleted while the backup is in progress, create a `space_id.del` file.                                      |
+    | `.ren`       | For a file that has been copied but renamed while the backup is in progress, create a space_id.ren file. This file should contain the new file name.                                                  |
+    | `.crpt`      | For files that cannot be fully copied due to encryption changes. These files will be recopied, and a `.crpt` file will be created for the existing file. |
+    | `.new.meta`  | Similar to `.new`, but for incremental backups.                                                                  |
+    | `.new.delta` | Similar to `.new`, but for incremental backups. Create `t1.ibd.meta` and `t1.ibd.delta` instead of `t1.ibd`.      |
+
+### Phase 3. Handling new metadata files in `--prepare`
+
+The following operations are performed to ensure data remains intact and consistent:
+{ .power-number }
+
+1. Remove the `.crpt` files that match the name after stripping the extension. This step is crucial before the `IBD` scan because these files are incomplete (they can be zero size).
+2. Perform a scan to create a mapping of `space_id` to file names.
+3. Delete the file matching the `space_id` using `space_id.del`. In case of incremental backups, also delete the corresponding `.new.meta` and `.new.delta` files.
+4. Rename the file matching the `space_id` to the name specified in the file using `space_id.ren`.
+5. Replace the file that matches the name without the `.new` extension using the `.new` extension.
+6. Replace the `meta` and `delta` files that match the name without the `.new` in the name using `.new.meta` and `.new.delta`.
+
+After `Phase 3` is completed, the regular crash recovery starts.
```

Comment:
Add a link to "How xtrabackup works" where it describe the "regular crash recovery"

Reply:
added